File: spring-framework/src/test/java/com/example/spring/framework/api/HelloController.java
package com.example.spring.framework.api;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

  public HelloController() {
    IO.println("hello");
  }

  @GetMapping(value = "/hello")
  public String helloBody() {
    return "Hello World!";
  }
}

File: htmx/src/main/kotlin/com/example/spring/htmx/api/HelloController.kt
package com.example.spring.htmx.api

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.ui.ModelMap
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable


@Controller
class HelloController {

  @GetMapping("/hello/{name}")
  fun hello(@PathVariable name: String, model: ModelMap): String {
    model["name"] = name
    return "hello"
  }

}